pkg/ddl/000032_iam_add_attach_namespace.up.sql

```diff
@@ -0,0 +1,45 @@
+BEGIN;
+
+-- wild_stars returns T if pattern matches hay using IAM-style wildcards: *
+-- is any string, ? is a single char.  It fails if pattern contains % or _
+-- chars.
+CREATE OR REPLACE FUNCTION pg_temp.wild_stars(pattern VARCHAR, hay VARCHAR)
+RETURNS BOOLEAN LANGUAGE plpgsql IMMUTABLE AS $$
+    DECLARE
+        unsafe BOOLEAN;
+	match BOOLEAN;
+
+    BEGIN
+        SELECT pattern LIKE '%\%%' OR pattern LIKE '%\_%' INTO STRICT unsafe;
+	IF unsafe THEN
+	    RAISE EXCEPTION 'unsafe pattern % contains "%%" or "_"', pattern;
+	END IF;
+        SELECT hay LIKE replace(replace(pattern, '*', '%'), '?', '_') INTO STRICT match;
+	RETURN match;
+    END;
+$$;
+
+-- jsonb_string translates a JSONB string object to SQL TEXT.  See
+-- https://stackoverflow.com/a/58755595.
+CREATE OR REPLACE FUNCTION pg_temp.jsonb_string(j JSONB) RETURNS TEXT LANGUAGE sql IMMUTABLE AS $$
+    SELECT j ->> 0
+$$;
+
+UPDATE auth_policies
```

Comment:
We don't do integration tests for migrations, but I'm assuming this was somehow tested nonetheless?

Reply:
Tested manually: I added manual policies and upgraded and downgraded.  Also upgraded multiple times to ensure idempotency.

You're correct, of course: we have no Postgres testing facility to run unit tests on these migration scripts.